src/Castle.Windsor/MicroKernel/DefaultKernel.cs

```diff
@@ -735,12 +735,17 @@ protected virtual void RegisterSubSystems()
 		}
 
 		protected object ResolveComponent(IHandler handler, Type service, IDictionary additionalArguments, IReleasePolicy policy)
+		{
+			return ResolveComponent(handler, service, additionalArguments, policy, ignoreParentContext: false);
+		}
+
+		private object ResolveComponent(IHandler handler, Type service, IDictionary additionalArguments, IReleasePolicy policy, bool ignoreParentContext)
```

Comment:
The `protected` method above can be deleted if you don't think users will have inherited from `DefaultKernel` and called `ResolveComponent(IHandler, Type, IDictionary, IReleasePolicy)` from inside.